sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/ParseDriver.scala

```diff
@@ -72,22 +72,41 @@ object ParseDriver extends Logging {
 
       // Find the non null token tree in the result.
       def nonNullToken(tree: CommonTree): CommonTree = {
-        if (tree.token != null || tree.getChildCount == 0) tree
-        else nonNullToken(tree.getChild(0).asInstanceOf[CommonTree])
+        if (tree.token != null || tree.getChildCount == 0) {
+          if (tree.token.getInputStream != null &&
+              tree.token.getInputStream.isInstanceOf[ANTLRStringStream]) {
+            val input = tree.token.getInputStream.asInstanceOf[ANTLRStringStream]
+            if (input.name == null) {
+              val callSite = org.apache.spark.util.Utils.getCallSite()
+              input.name = callSite.shortForm
+            }
+          } else {
+            val input = new ANTLRStringStream(command)
+            val callSite = org.apache.spark.util.Utils.getCallSite()
+            input.name = callSite.shortForm
+            tree.token.setInputStream(input)
+          }
+          tree
+        } else nonNullToken(tree.getChild(0).asInstanceOf[CommonTree])
```

Comment:
I agree. I will call Utils.getCallSite() in a constructor of ASTNode and revert changes in ParseDriver.scala.